codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionDecoder.java

```diff
@@ -322,9 +308,8 @@ public void onHeadersRead(ChannelHandlerContext ctx, int streamId, Http2Headers
         public void onPriorityRead(ChannelHandlerContext ctx, int streamId, int streamDependency, short weight,
                 boolean exclusive) throws Http2Exception {
             Http2Stream stream = connection.stream(streamId);
-            verifyGoAwayNotReceived();
-            if (shouldIgnoreFrame(stream, true)) {
-                // Ignore this frame.
+
+            if (stream != null && streamCreatedAfterGoAwaySent(stream)) {
```

Comment:
Can we take advantage of the below check on `stream == null`?  Add an `else if(streamCreatedAfterGoAwaySent(stream)) { return; }`  This will cut down on the duplicate conditionals, and I don't think sacrifices too much clarity.